presto-main/src/main/java/com/facebook/presto/util/MergeSortedPages.java

```diff
@@ -143,9 +144,11 @@ private MergeSortedPages() {}
         @Override
         public ProcessorState<Page> process(Optional<PageWithPosition> pageWithPositionOptional)
         {
-            if (yieldSignal.isSet()) {
+            if (yieldSignal.isSet() && !lastComputeYielded) {
+                lastComputeYielded = true;
                 return ProcessorState.yield();
             }
+            lastComputeYielded = false;
```

Comment:
as you @sopel39 proposed -- this can be implemented in a generic way, by a wrapper

Reply:
> as you @sopel39 proposed -- this can be implemented in a generic way, by a wrapper

I did it this way. You could have a look if you want